Upgrade code to be Symfony 6 compatible

MKodde opened the pull request:
### What changed here?

- Travis was disconnected, and QA tests now run on GHA
- Code was upgraded to run on Symfony 6.3 (as soon as 6.4 releases we should upgrade to that)
- Code was upgraded to run on PHP8, certain PHP8 features have been implemented throughout the code
- Strict types have been enforced
- PHPStan has been added to the QA tooling and the highest standard is enabled with a minimum of baseline entries
- Config folder has been simplified. Overridable config is now placed in /config/openconext
- Front end assets have been moved from /public folder to the /assets folder

parijke commented on config/packages/security.yaml:
```diff
@@ -1,22 +1,15 @@
 security:
+    enable_authenticator_manager: true
```
Dit hoeft dacht ik niet meer vanaf 6.4 en is deprecated

MKodde replied:
Good point, removed that.

parijke commented on config/routes.yaml:
```diff
@@ -1,5 +1,2 @@
-#index:
-#    path: /
-#    controller: App\Controller\DefaultController::index
 gssp_saml:
```
Ik heb routing vervangen in SelfService door #[Route()] attribute

MKodde replied:
This opened a can of worms ;)

I already defined the attribute based routes everywhere. But the app is not loading them appropriately. Thats why all the extra config is added. I can not figure out how to configure this correctly. I'm putting this on the backlog for now. But a thing to fix for surre!

parijke commented on dev/Controller/SPController.php:
```diff
@@ -42,28 +40,15 @@
-        ServiceProvider $serviceProvider,
-        IdentityProvider $identityProvider,
-        PostBinding $postBinding
+        private readonly ServiceProvider $serviceProvider,
```
Volgens mij kan de class dan readonly ipv alle props

MKodde replied:
Given we extend the abstract controller, this is not an option. Maybe we should stop using that abstract controller.. Also one for the backlog. But given this is a dev resource. I'm not super concerned at this point.

parijke commented on dev/Controller/SPController.php:
```diff
@@ -42,28 +40,15 @@
-     * @throws \Exception
-     */
-    public function demoSpAction(Request $request)
+    #[Route(path: '/demo/sp', name: 'sp_demo')]
```
ook attrubute gebruikt, top

MKodde replied:
Took some getting used to, but in the end I much prefer the new attribute based annotations. More YAMLy FTW

parijke commented on dev/Controller/SPController.php:
```diff
@@ -42,28 +40,15 @@
-     */
-    public function demoSpAction(Request $request)
+    #[Route(path: '/demo/sp', name: 'sp_demo')]
+    public function demoSp(Request $request): Response|RedirectResponse
```
RedirectReposne = response, dus die kan weg als return type

MKodde replied:
That's correct, but to scratch PHPStan's itch this was added to make the return type definition as specific as possible.

parijke commented on dev/Controller/SPController.php:
```diff
@@ -140,12 +120,8 @@ public function assertionConsumerServiceAction(Request $request)
-     * @return string
      */
-    private function toFormattedXml($xml)
+    private function toFormattedXml(string|bool $xml): string|bool
```
Kan hier $xml de waarde true/false hebben?

MKodde replied:
Yes it can, we directly pass the contents of the `base64_decode` function into this method. Im going to add a guard to prevent this. And then the type hint can be more strict! Good point mate!

parijke reviewed (commented):
In de regel prima lijkt me. Wat kleine suggesties / vragen

MKodde commented on dev/Controller/SPController.php:
```diff
@@ -193,19 +160,12 @@ private function loadServiceProviderPrivateKey()
         libxml_disable_entity_loader($previous);
 
-        return Response::fromXML($asXml->documentElement);
+        return \SAML2\Response::fromXML($asXml->documentElement);
```
Must have slipped my attention! Good call

MKodde commented on src/Surfnet/Gssp/Infrastructure/Controller/DefaultController.php:
```diff
@@ -24,26 +26,21 @@
     public function __construct(
-        AuthenticationService $authenticationService,
-        RegistrationService $registrationService
+        private readonly AuthenticationService $authenticationService,
```
Same as before. Unable to do this due to the use of the abstract controller

MKodde commented on src/Surfnet/Gssp/Infrastructure/Controller/DefaultController.php:
```diff
@@ -24,26 +26,21 @@
-        AuthenticationService $authenticationService,
-        RegistrationService $registrationService
+        private readonly AuthenticationService $authenticationService,
+        private readonly RegistrationService $registrationService
```
Have you found a way to automate this? I'll try to fix this for all classes in one go..

:crossed_fingers:

parijke replied:
Great setting, changed that as well!
![image](https://github.com/OpenConext/Stepup-gssp-example/assets/7671998/47f0117d-f84e-4e7a-baeb-001e250d6fad)

MKodde commented on src/Surfnet/Gssp/Infrastructure/EventSubscriber/LocaleResponseListener.php:
```diff
@@ -66,20 +62,15 @@ public function onKernelRequest(RequestEvent $event)
-            '/',
-            $this->getNakedDomain()
-        );
+        $cookie = Cookie::create(self::STEPUP_LOCALE_COOKIE, $request->getLocale(), 0, '/', $this->getNakedDomain());
```
That's mainly: I'm not gonna change what ain't broke

MKodde commented:
Thanks for the review @parijke 

I addressed your concerns to the best of my abilities. I did reject some as they added too little value for the additional effort. I tried to explain why in comments. Can you re-evaluate if you can approve these changes?
